### PR TITLE
feat(sync): sync and pair 1-1 chats and groups

### LINKED ISSUE #10324: When a user performs a fresh install and restores via seed phrase, 1on1 and ad-hoc group chat conversations that happened in the last 30 days should be restored
Version 0.11, Window 10

To reproduce:

1. Perform a fresh install of Status Desktop, and restore an account that's been active in the last 30 days via Seed Phrase

2. When the user lands inside Status after restoring, all of the user's contacts have been successfully restored from the backup to waku (Yay! :-) ).  This can be checked by navigating to Settings/Messaging/Contacts and viewing the restored contacts

3. Navigate to the Messages section of the app.

--- What currently incorrectly happens ---

4. The Messages secondary nav will be empty, with no 1on1 and ad-hoc group chats listed

--- Desired functionality ---

4. All 1on1 and ad-hoc group chats that have had activity in them in the last 30 days should be restored.  By 'restored', I mean that they should be listed in the Messages section secondary nav, in recency order (e.g. the chat with the most recent message at the top, the chat with the second most recent message 2nd from the top, etc..).  I think that due to 'perfect forward secrecy' the messages themselves in the 1on1 and ad-hoc group chats cannot be restored via fetching the user's last 30 days of messages from a waku mailserver (but I would love to be proven wrong about this!), but at the minimum we should restore the list of 1on1 and ad-hoc group chats into the Messages section secondary nav.

One way to implement this would be to simply add a list of all 1on1 and ad-hoc group chats in the Messages section in recency to the list of user data that's: 1) backed up to Waku every 8 hours and 2) restored to a new device during initial pairing in the device syncing flow.

--------------------

This issue is a more detailed write up of **Issue 4.** in the following github issue https://github.com/status-im/status-desktop/issues/10276

Assigning to Milestone 0.13, but feel free to pull into Milestone 0.12 if there is time.

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #10324 and https://github.com/status-im/status-go/issues/3547

status-go PRs:
- https://github.com/status-im/status-go/pull/4073
- https://github.com/status-im/status-go/pull/4102

[import-chats.webm](https://github.com/status-im/status-go/assets/11926403/0c241a66-6ee6-4f1a-8f9d-b699e20ba5d3)
[pairing-chats.webm](https://github.com/status-im/status-go/assets/11926403/a263bf2e-b7cc-4b17-9043-c51a565b4d9e)

### CODE REVIEW

**Review by saledjenic — APPROVED** [review]

**Comment by jrainville** [comment]
@anastasiyaig can you have someone test this PR? 

It makes it possible to sync and pair 1-1 chats and group chats. It would be good to test with mobile and desktop.

**Comment by anastasiyaig** [comment]
sure, will do that tomorrow first thing @jrainville

**Review by mprakhov — APPROVED** [review]

**Comment by anastasiyaig** [comment]
@cammellos do you know if this change has been implemented on mobile so i can use develop or is there any special PR ?

**Comment by anastasiyaig** [comment]
alright, so we want to sync only metadata here, not messages itself. @lukaszso please test this

**Review by lukaszso — APPROVED** [review]
Looks like it works fine. Sometimes it takes longer to pull the chat list but looks like it's not related to this PR as the time is similar to fetching community messages on master. 
TL;DR LGTM.
